src/celeritas/user/ActionTimes.cc

```diff
@@ -0,0 +1,66 @@
+//------------------------------- -*- C++ -*- -------------------------------//
+// Copyright Celeritas contributors: see top-level COPYRIGHT file for details
+// SPDX-License-Identifier: (Apache-2.0 OR MIT)
+//---------------------------------------------------------------------------//
+//! \file celeritas/user/ActionTimes.cc
+//---------------------------------------------------------------------------//
+#include "ActionTimes.hh"
+
+#include "corecel/sys/ActionRegistry.hh"
+
+namespace celeritas
+{
+//---------------------------------------------------------------------------//
+/*!
+ * Construct from ID and actions.
+ */
+ActionTimes::ActionTimes(AuxId aux_id, SPActionRegistry const& action_reg)
+    : aux_id_(aux_id), action_reg_(action_reg)
+{
+    CELER_EXPECT(aux_id_);
+    CELER_EXPECT(action_reg_);
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Build core state data for a stream.
+ */
+auto ActionTimes::create_state(MemSpace, StreamId, size_type) const -> UPState
+{
+    return std::make_unique<ActionTimesState>();
+}
+
+//---------------------------------------------------------------------------//
+/*!
+ * Access the state.
+ */
+ActionTimesState const& ActionTimes::state(AuxStateVec const& aux) const
+{
+    return dynamic_cast<ActionTimesState const&>(aux.at(aux_id_));
```

Comment:
In `AuxStateVec.hh` there's a `get` method intended to reduce boilerplate like this and with more debugging help:

```suggestion
    return get<ActionTimesState>(aux,aux_id_);
```

Reply:
Oh right, forgot about that one!